internal/services/signalr/web_pubsub_hub_resource.go

```diff
@@ -105,6 +106,46 @@ func resourceWebPubSubHub() *pluginsdk.Resource {
 				},
 			},
 
+			"event_listener": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"user_event_name_filter": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+							},
+						},
+
+						"system_event_name_filter": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{
+									"connected",
+									"disconnected",
+								}, false),
+							},
+						},
+
+						"eventhub_namespace_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: eventhubValidate.ValidateEventHubNamespaceName(),
+						},
+
+						"eventhub_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: eventhubValidate.ValidateEventHubName(),
+						},
```

Comment:
These can be condensed down to `eventhub_id`

```suggestion
						"eventhub_id": {
							Type:         pluginsdk.TypeString,
							Required:     true,
							ValidateFunc: eventhubs.ValidateEventhubID,
						},
```

Reply:
Thanks @stephybun , we can't use eventhub_id as service only return the eventhub name and eventhub namespace name back, it's hard to put the id back using only the name (resource group info can't be got)